CORRELATION_CONTEXT_HTTP_HEADER_FORMAT.md

```diff
@@ -0,0 +1,73 @@
+# Correlation Context HTTP Header Format
+
+A correlation context header is used to pass the name-value context properties for the trace. This is a companion header for the `Trace-Context`. The values should be passed along to any child requests. Note that uniqueness of the key within the `Correlation-Context` is not guaranteed. Context received from upstream service may be altered before passing it along.
+
+# Format
+
+## Header name
+
+`Correlation-Context`
+
+Multiple correlation context headers are allowed. Values can be combined in a single header according to [rfc](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2).
+
+## Field value
+
+`name1=value1[;type=(number|bool|string)], name2=value2[;type=(number|bool|string)]`
+
+**Limits:**
+1. Maximum number of name-value pairs: `180`.
+2. Maximum number of bytes per a single name-value pair: `4096`.
+3. Maximum total length of all name-value pairs: `8192`.
+
+## Name format
+
+Url encoded string. Spaces should be trimmed from beginning and the end of the name. Names are case sensitive.
+
+## Supported value types
+
+All spaces should be trimmed from the beginning and the end of the value.
+
+### String
+
+Default value type. Represents a url encoded string value. Value is case sensitive.
+
+**Examples**:
+
+```
+component=Frontend
+component=Frontend;type=string
```

Comment:
this will explode into more bytes a lot. I like that type defaults to string to reduce that. unless you are expecting to add more attributes, possibly better a keyword approach ex priority=0.1;number

Reply:
or even cost=42;n, worthless=0;b

Reply:
or support strings only. HTTP is a text format so the value is going to be read as a string anyway, and the application that's interested in interpreting a particular baggage key should already know which type it should be.

Reply:
Prefer String only.

Reply:
Next thing I was thinking people may ask is the ttl. For context properties like "component name". Another it PII (personal identifiable information). PII flag may help to identify when to pass it along and when not to. Like pass it to the dependent services with the this host name, but not to others.

Alternative to `;type=` may be some sort of prefix for a value. The only problem with prefix as a custom escape scheme and ultimately a conflict with the human readability requirement.

Reply:
I'm open to ideas and ok to strip out the `type`

Reply:
Does any tracer process other tracer's `Correlation-Context` HEAD? Do we have any use cases? Or are we going to define some common keys? @SergeyKanzhelev @adriancole

Reply:
@wu-sheng I was thinking defining the well-known names later as a separate discussion. Same way open tracing defines specific tags

Reply:
@SergeyKanzhelev Let's discuss that later. In my experiences, the contents in `Correlation-Context` are really much implementation related. Hard to say common keys are useful or not.